src/org/zaproxy/zap/extension/encoder2/EncodeDecodeDialog.java

```diff
@@ -524,6 +550,19 @@ private void updateEncodeDecodeFields() {
 			illegalUTF84ByteField.setText("");
 		}
 
+        try {
+            text2UnicodeField.setText(getEncoder().getUnicodeEncode(getInputField().getText()));
+        } catch (final Exception e) {
+            // Not unexpected
+            text2UnicodeField.setText("something went wrong.");
+        }
+
+        try {
+            unicode2TextField.setText(getEncoder().getUnicodeDecode(getInputField().getText()));
+        } catch (final Exception e) {
+            // Not unexpected
+            unicode2TextField.setText("something went wrong.");

```

Comment:
Should have a leading capital.

Reply:
- The message should be more informative (log the exception, debug/warn?) and should be internationalised, it's being shown to the user.
 - Might worth disable the text field as it's more obvious that something is wrong (base64 does that when there's an error in the input).